module/data/advancement/scale-value.mjs

```diff
@@ -1,17 +1,325 @@
+   * @returns {object|null}
+   */
+  static convertFrom(original) {
+    return { value: original.formula };
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * This scale value prepared to be used in roll formulas.
+   * @type {string|null}
+   */
+  get formula() { return this.value; }
+
+  /* -------------------------------------------- */
+
+  /**
+   * This scale value formatted for display.
+   * @type {string|null}
+   */
+  get display() { return this.formula; }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Shortcut to the prepared value when used in roll formulas.
+   * @returns {string}
+   */
+  toString() {
+    return this.formula;
+  }
+}
+
+
+/**
+ * Scale value data type that stores numeric values.
+ *
+ * @property {number} value  Numeric value.
+ */
+export class ScaleValueTypeNumber extends ScaleValueType {
+
+  /** @inheritdoc */
+  static defineSchema() {
+    return {
+      value: new foundry.data.fields.NumberField({required: true})
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  static get metadata() {
+    return foundry.utils.mergeObject(super.metadata, {
+      label: "DND5E.AdvancementScaleValueTypeNumber",
+      hint: "DND5E.AdvancementScaleValueTypeHintNumber",
+      isNumeric: true
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  static convertFrom(original) {
+    const value = Number(original.formula);
+    if ( Number.isNaN(value) ) return null;
+    return { value };
+  }
+}
+
+
+/**
+ * Scale value data type that stores challenge ratings.
+ *
+ * @property {number} value  CR value.
+ */
+export class ScaleValueTypeCR extends ScaleValueTypeNumber {
+
+  /** @inheritdoc */
+  static defineSchema() {
+    return {
+      value: new foundry.data.fields.NumberField({required: true, min: 0})
+      // TODO: Add CR validator
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  static get metadata() {
+    return foundry.utils.mergeObject(super.metadata, {
+      label: "DND5E.AdvancementScaleValueTypeCR",
+      hint: "DND5E.AdvancementScaleValueTypeHintCR"
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  get display() {
+    switch ( this.value ) {
+      case 0.125: return "&frac18;";
+      case 0.25: return "&frac14;";
+      case 0.5: return "&frac12;";
+      default: return super.display;
+    }
+  }
 }
+
+
+/**
+ * Scale value data type that stores dice values.
+ *
+ * @property {number} n    Number of dice.
+ * @property {number} die  Die face.
+ */
+export class ScaleValueTypeDice extends ScaleValueType {
+
+  /** @inheritdoc */
+  static defineSchema() {
+    return {
+      number: new foundry.data.fields.NumberField({nullable: true, integer: true, positive: true}),
+      face: new foundry.data.fields.NumberField({required: true, integer: true, positive: true})
```

Comment:
`DiceTerm` uses the plural `faces` (e.g., an 8-sided die has 8 faces on the polyhedron). I think the singular `face`, probably does not make sense in this context.